contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -1609,3 +1609,24 @@ Example: "must be greater than \`request\`".
 be less than 256", "must be greater than or equal to 0".  Do not use words
 like "larger than", "bigger than", "more than", "higher than", etc.
 * When specifying numeric ranges, use inclusive ranges when possible.
+
+
+## Automatic Resource Allocation And Deallocation
+
+API objects often are union object contains the following:
+1. One or more fields identifying `Type` specific to API object.
+3. A union of all the fields that an object can represent.
+
+Controllers operating on the API type often allocate resources based on 
+the `Type` and/or some additional data provided by user. A canonical example 
+of this is the `Service` API object where resources such as IPs and network ports 
+will be set in the API object based on `Type` and allocation is performed 
+automatically when the user does not specify resources or reservation when the 
```

Comment:
add a comma before the  "or" ?  Or maybe reflow as something like

"based on `Type`. When the user does not specify resources, they will be allocated, and when the user specifies exact value, they will be reserved or rejected. "